### DISCUSSION TITLE
Single quote only for characters

### DISCUSSION THREAD

**Discussion by david-ragazzi** [modular/modular#731] [General]
Hi,

Playing in the JupyterLab, I see single quotes work both single characters and whole strings. Is there a future plan to Mojo follow a more strict use for single quote delimiting it to represent only characters like most system languages like C/C++, Rust, etc? For instance:

`var my_string = String("ABCDEF")`
`my_string[0] = 'X'  # Single quote would be only for character type`
`my_string[1] = "Y" # Error here`

The last row will raise an error because compiler will interpret "Y" as a string, not a character as required by string's set_item() method.

This also would remove ambiguity where code use both single and double quotes to represent strings and characters. Instead of:

`var str1 = "Some string"`
`var str2 = 'Another string'`
`var ch1 = "A"`
`var ch2 = 'B'`

we would have:

`var str1 = "Some string"`
`var str2 = "Another string"`
`var ch1 = 'A'`
`var ch2 = 'B'`

PS: Yes, I know PEP states that you can use both, but maybe it's time to modernize PEP to uniform the representation of strings.

**Comment by gryznar** [comment]
How about situation in which string contains `"`? I am fine with this proposition if implementation does not enforce to escape every `"` char separately with `\`.

**Reply by david-ragazzi** [reply]
It's not always raibows and butterfies... But I think there are more pro than cons...

By the way, if you need use `'` in a string where your code use `'` as string demilitador, you will have the same problem even in Python. Unless you don't mind having inconsistent code where both single and double quotes are used for demilitate strings, which even PEP is not favorable.

**Reply by gryznar** [reply]
You are right
